diff --color -ruN -x .idea -x cmake-build-debug SuiteSparse-7.3.1/CHOLMOD/CMakeLists.txt SuiteSparse-7.3.1.patched/CHOLMOD/CMakeLists.txt
--- SuiteSparse-7.3.1/CHOLMOD/CMakeLists.txt	2023-11-03 23:42:14.000000000 +0100
+++ SuiteSparse-7.3.1.patched/CHOLMOD/CMakeLists.txt	2026-02-16 14:47:25.726189383 +0100
@@ -124,10 +124,10 @@
 
     option ( NCHECK "ON: do not use Check Module.  OFF (default): use Check Module" off )
 
-    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Check )
-        # Check module does not appear so don't use it
-        set ( NCHECK true )
-    endif ( )
+#    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Check )
+#        # Check module does not appear so don't use it
+#        set ( NCHECK true )
+#    endif ( )
     if ( NCHECK )
         # if NCHECK is true: do not build the Check module
         add_compile_definitions ( NCHECK )
@@ -139,10 +139,10 @@
 
     option ( NMATRIXOPS "ON: do not use MatrixOps Module.  OFF (default): use MatrixOps Module" off )
 
-    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/MatrixOps )
-        # MatrixOps module does not appear so don't use it
-        set ( NMATRIXOPS true )
-    endif ( )
+#    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/MatrixOps )
+#        # MatrixOps module does not appear so don't use it
+#        set ( NMATRIXOPS true )
+#    endif ( )
     if ( NMATRIXOPS )
         # if NMATRIXOPS is true: do not build the Check module
         add_compile_definitions ( NMATRIXOPS )
@@ -154,10 +154,10 @@
 
     option ( NCHOLESKY "ON: do not use Cholesky Module.  OFF (default): use Cholesky Module" off )
 
-    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Cholesky )
-        # Cholesky module does not appear so don't use it
-        set ( NCHOLESKY true )
-    endif ( )
+#    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Cholesky )
+#        # Cholesky module does not appear so don't use it
+#        set ( NCHOLESKY true )
+#    endif ( )
     if ( NCHOLESKY )
         # if NCHOLESKY is true: do not build the Cholesky module, and do not
         # build the Supernodal or Modify modules that depend on it.
@@ -172,10 +172,10 @@
 
     option ( NMODIFY "ON: do not use Modify Module.  OFF (default): use Modify Module" off )
 
-    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Modify )
-        # Modify module does not appear so don't use it
-        set ( NMODIFY true )
-    endif ( )
+#    if ( NOT EXISTS ${CMAKE_SOURCE_DIR}/Modify )
+#        # Modify module does not appear so don't use it
+#        set ( NMODIFY true )
+#    endif ( )
     if ( NMODIFY )
         # if NMODIFY is true: do not build the Modify module
         add_compile_definitions ( NMODIFY )
diff --color -ruN -x .idea -x cmake-build-debug SuiteSparse-7.3.1/CMakeLists.txt SuiteSparse-7.3.1.patched/CMakeLists.txt
--- SuiteSparse-7.3.1/CMakeLists.txt	1970-01-01 01:00:00.000000000 +0100
+++ SuiteSparse-7.3.1.patched/CMakeLists.txt	2026-02-16 13:57:03.619447476 +0100
@@ -0,0 +1,37 @@
+cmake_minimum_required(VERSION 3.20)
+project(SparseSuite)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/SuiteSparse_config/cmake_modules")
+
+# avoid blas check as it is hard to compile on windows
+# and that we don't need it for the suitesparse modules we need (amd, colamd, btf, klu and cxsparse)
+set ( SuiteSparse_BLAS_integer "int32_t" )
+set ( BLAS_FOUND true )
+
+add_subdirectory(SuiteSparse_config/)
+add_library(SuiteSparse::SuiteSparseConfig ALIAS SuiteSparseConfig)
+add_library(SuiteSparse::SuiteSparseConfig_static ALIAS SuiteSparseConfig_static)
+
+add_subdirectory(AMD)
+add_library(SuiteSparse::AMD ALIAS AMD)
+add_library(SuiteSparse::AMD_static ALIAS AMD_static)
+
+add_subdirectory(COLAMD)
+add_library(SuiteSparse::COLAMD ALIAS COLAMD)
+add_library(SuiteSparse::COLAMD_static ALIAS COLAMD_static)
+
+add_subdirectory(BTF)
+add_library(SuiteSparse::BTF ALIAS BTF)
+add_library(SuiteSparse::BTF_static ALIAS BTF_static)
+
+add_subdirectory(KLU)
+add_library(SuiteSparse::KLU ALIAS KLU)
+add_library(SuiteSparse::KLU_static ALIAS KLU_static)
+
+add_subdirectory(CXSparse)
+add_library(SuiteSparse::CXSparse ALIAS CXSparse)
+add_library(SuiteSparse::CXSparse_static ALIAS CXSparse_static)
+
+add_subdirectory(CHOLMOD)
+add_library(SuiteSparse::CHOLMOD ALIAS CHOLMOD)
+add_library(SuiteSparse::CHOLMOD_static ALIAS CHOLMOD_static)
\ No newline at end of file
diff --color -ruN -x .idea -x cmake-build-debug SuiteSparse-7.3.1/SuiteSparse_config/CMakeLists.txt SuiteSparse-7.3.1.patched/SuiteSparse_config/CMakeLists.txt
--- SuiteSparse-7.3.1/SuiteSparse_config/CMakeLists.txt	2023-11-03 23:42:14.000000000 +0100
+++ SuiteSparse-7.3.1.patched/SuiteSparse_config/CMakeLists.txt	2026-02-16 13:52:53.862282209 +0100
@@ -62,7 +62,10 @@
     find_package ( OpenMP )
 endif ( )
 
-include ( SuiteSparseBLAS )
+## patch CMakeLists.txt of SuiteSparse_config module to avoid blas check as it is hard to compile on windows
+## and that we don't need it for the suitesparse modules we need (amd, colamd, btf, klu and cxsparse)
+set ( SuiteSparse_BLAS_integer "int32_t" )
+set ( BLAS_FOUND true )
 
 #-------------------------------------------------------------------------------
 # configure files
